backend/provisioner/deployment/deployment.go

```diff
@@ -0,0 +1,133 @@
+package deployment
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1beta1/provisioner"
+)
+
+type TaskState string
+
+const (
+	TaskStatePending TaskState = ""
+	TaskStateRunning TaskState = "running"
+	TaskStateDone    TaskState = "done"
+	TaskStateFailed  TaskState = "failed"
+)
+
+// Task is a unit of work for a deployment
+type Task struct {
```

Comment:
Should all these be public? Generally I think unless a type is purely data it's preferable to have private fields and a constructor, to make it very clear what is and isn't required at construction time.